src/ui/keymap_popup.rs

```diff
@@ -0,0 +1,106 @@
+use ratatui::{
+    backend::CrosstermBackend,
+    layout::{Alignment, Rect},
+    text::Text,
+    widgets::{Block, Borders, Clear, Padding, Paragraph},
+    Frame,
+};
+
+use super::theme::Theme;
+
+include!(concat!(env!("OUT_DIR"), "/keybindings.rs"));
+
+pub struct KeymapPopup {
+    visible: bool,
+    scroll_y: u16,
+    scroll_x: u16,
+    content: Text<'static>,
+}
+
+impl KeymapPopup {
+    pub fn new() -> Self {
+        Self {
+            visible: false,
+            scroll_y: 0,
+            scroll_x: 0,
+            content: Text::from(KEYBINDINGS_TABLE),
+        }
+    }
+
+    pub fn toggle(&mut self) {
+        self.visible = !self.visible;
+        if self.visible {
+            self.scroll_y = 0;
+            self.scroll_x = 0;
+        }
+    }
+
+    pub fn go_down(&mut self) {
+        self.scroll_y = self.scroll_y.saturating_add(1).min(KEYBINDINGS_LEN);
+    }
+
+    pub fn go_up(&mut self) {
+        self.scroll_y = self.scroll_y.saturating_sub(1);
+    }
+
+    pub fn go_right(&mut self) {
+        self.scroll_x = self.scroll_x.saturating_add(1).min(KEYBINDINGS_LINE_LEN);
+    }
+
+    pub fn go_left(&mut self) {
+        self.scroll_x = self.scroll_x.saturating_sub(1);
+    }
+
+    pub fn draw(&self, frame: &mut Frame<CrosstermBackend<std::io::Stdout>>, theme: &dyn Theme) {
+        if !self.visible {
+            return;
+        }
+
+        let popup_area = Self::get_popup_area(frame.size());
+
+        let max_y = KEYBINDINGS_LEN.saturating_sub(popup_area.height - 2);
+        let scroll_y = self.scroll_y.min(max_y);
+        let max_x = KEYBINDINGS_LINE_LEN.saturating_sub(popup_area.width - 4);
+        let scroll_x = self.scroll_x.min(max_x);
+
+        let paragraph = Paragraph::new(self.content.clone())
+            .block(
+                Block::default()
+                    .borders(Borders::ALL)
+                    .border_style(theme.search_popup_border())
+                    .title(concat!(
+                        "Keybindings - ",
+                        env!("CARGO_PKG_NAME"),
+                        " ",
+                        env!("CARGO_PKG_VERSION"),
+                    ))
+                    .title_alignment(Alignment::Center)
+                    .padding(Padding::horizontal(1)),
```

Comment:
```suggestion
                    .padding(Padding::uniform(1)),
```
IMO it looks better then. You need to change the calculation of vertical size then.

Reply:
Done